### PR TITLE
Make dns_servers optional in subnets

### PR DESCRIPTION
**Pull request by voidspace**
Make dns_servers optional in the subnets response from MAAS 2.

### CODE REVIEW

File: subnet_test.go [hunk 1]
```diff
@@ -86,7 +86,6 @@ var subnetResponse = `
         "space": "space-0",
         "id": 34,
         "resource_uri": "/MAAS/api/2.0/subnets/34/",
-        "dns_servers": [],
```
Comment (dimitern) [inline subnet_test.go:89]: How about explicitly using `"dns_server": null` above instead? Can it be null coming from MAAS or it's just omitted?

Comment (voidspace) [inline subnet_test.go:89]: I've checked the output - and the problematic one _is_ "null", so I've updated the test.

**Comment by dimitern** [comment]
LGTM, thanks!

**Comment by voidspace** [comment]
$$merge$$

**Pull request merged**